test/sanity/code-smell/update-bundled.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# (c) 2018, Ansible Project
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+
+import glob
+import json
+import os.path
+import sys
+from distutils.version import LooseVersion
+
+import packaging.specifiers
+
+from ansible.module_utils.urls import open_url
+
+
+def compat_libs(basedir):
+    for filename in glob.glob(os.path.join(basedir, 'lib/ansible/compat/*/__init__.py')):
+        if 'compat/tests' in filename:
+            # compat/tests doesn't bundle any code
+            continue
+        yield filename
+
+    yield os.path.join(basedir, 'lib/ansible/module_utils/distro/__init__.py')
+    yield os.path.join(basedir, 'lib/ansible/module_utils/six/__init__.py')
+    yield os.path.join(basedir, 'lib/ansible/module_utils/compat/ipaddress.py')
+    # backports.ssl_match_hostname should be moved to its own file
+    yield os.path.join(basedir, 'lib/ansible/module_utils/urls.py')
+
+
+def get_bundled_metadata(filename):
+    with open(filename, 'r') as module:
+        for line in module:
+            if line.strip().startswith('_BUNDLED_METADATA'):
+                data = line[line.index('{'):].strip()
+                break
+        else:
+            raise Exception('{0} contained no metadata.  Could not check for updates'.format(filename))
```

Comment:
Instead of raising and exception, print the error in the format: `path: message`

Reply:
This exception is caught further down.

Reply:
The catch is here: https://github.com/ansible/ansible/pull/51759/files#diff-36a6eef1b3c45cf98f385ac3a81b0cadR83